corehq/util/model_log.py

```diff
@@ -1,45 +1,59 @@
 from enum import Enum
 
-from django.contrib.admin.models import ADDITION, CHANGE, DELETION, LogEntry
-from django.contrib.admin.options import get_content_type_for_model
-from django.utils.encoding import force_text
+from corehq.apps.users.models_sql import HQLogEntry, UpdateDetails
+
+MODEL_PROPERTIES_TO_IGNORE = [
+    '_id', '_rev', 'reporting_metadata', 'password',
+    'devices', 'last_device', 'device_ids'
+]
 
 
 class ModelAction(Enum):
-    CREATE = ADDITION
-    UPDATE = CHANGE
-    DELETE = DELETION
+    CREATE = HQLogEntry.CREATE
+    UPDATE = HQLogEntry.UPDATE
+    DELETE = HQLogEntry.DELETE
 
 
-def log_model_change(user, model_object, message=None, fields_changed=None, action=ModelAction.UPDATE):
+def log_model_change(domain, user, model_object, message=None, fields_changed=None, action=ModelAction.UPDATE,
+                     can_skip_domain=False, changed_via=None):
     """
-    :param user: User making the change (couch user or django user)
-    :param model_object: The object being changed (must be a Django model)
-    :param message: Message text
-    :param fields_changed: List of model field names that have
+    :param domain: domain where the update was initiated
+    :param user: User making the change (couch user) or SYSTEM_USER_ID
+    :param model_object: The user being changed (couch user)
+    :param message: Optional Message text
+    :param fields_changed: dict of model fields that have changed with current value
     :param action: Action on the model
+    :param can_skip_domain: flag to allow domain less entry
+    :param changed_via: changed via medium i.e API/Web
     """
-    from corehq.apps.users.models import CouchUser
-    if isinstance(user, CouchUser):
-        user = user.get_django_user()
-
-    if message is None and fields_changed is None:
-        raise ValueError("One of 'message' or 'fields_changed' is required.")
+    from corehq.apps.users.util import SYSTEM_USER_ID
 
-    if message is not None and fields_changed is not None:
-        raise ValueError("'message' and 'fields_changed' are mutually exclusive")
+    if not domain and user != SYSTEM_USER_ID and not can_skip_domain:
+        raise ValueError("Please pass domain")
 
-    if message is None:
+    if action == ModelAction.UPDATE:
         if not fields_changed:
-            raise ValueError("'fields_changed' must not be empty")
+            raise ValueError("'fields_changed' is required for update.")
+
+    return HQLogEntry.objects.create(
+        domain=domain,
+        object_type=model_object.doc_type,
+        object_id=model_object.get_id,
+        by_user_id=SYSTEM_USER_ID if user == SYSTEM_USER_ID else user.get_id,
+        details=UpdateDetails.wrap({
+            'changes': _get_change_details(model_object, action, fields_changed),
+            'changed_via': changed_via,
+        }),
```

Comment:
I don't see much value in wrapping this here since you're just making a dict
```suggestion
        details={
            'changes': _get_change_details(model_object, action, fields_changed),
            'changed_via': changed_via,
        },
```